src/lib/provable/bigint.ts

```diff
@@ -88,6 +94,10 @@ function createProvableBigInt(modulus: bigint, config?: BigIntParameter) {
     modulus < config_.max,
     `ProvableBigInt: modulus exceeds the max supported size of 2^${config_.max}`
   );
+  assert(
```

Comment:
Let's enforce these checks on the type level, please. 

Given that these values cannot be dynamic, type level checks will make it easier for the developer to use this API in a type safe way without unexpected runtime errors - it's also less error prone and gives better readability imo